app/controllers/api/settings_controller.rb

```diff
@@ -1,22 +1,39 @@
 module Api
   class SettingsController < BaseController
     def index
-      render_resource :settings, slice_settings(exposed_settings)
+      render_resource :settings, whitelist_settings(settings_hash)
     end
 
     def show
-      raise NotFoundError, "Settings category #{@req.c_id} not found" unless exposed_settings.include?(@req.c_id)
-      render_resource :settings, slice_settings(@req.c_id)
+      settings_value = entry_value(whitelist_settings(settings_hash), @req.c_suffix)
+
+      raise NotFoundError, "Settings entry #{@req.c_suffix} not found" if settings_value.nil?
+      render_resource :settings, settings_entry_to_hash(@req.c_suffix, settings_value)
     end
 
     private
 
-    def exposed_settings
-      ApiConfig.collections[:settings][:categories]
+    def whitelist_settings(settings)
```

Comment:
This seems (a) sufficiently complicated, and (b) already abstracted away from details of the request to warrant a service object that could be perhaps better tested in isolation. WDYT?